sql/2020/20_Caching/appcache_and_serviceworkers_2019.sql

```diff
@@ -0,0 +1,17 @@
+#standardSQL
+# Use of AppCache and ServiceWorkers
+SELECT
+  IF(STARTS_WITH(url, 'https'), 'https', 'http') AS http_type,
+  JSON_EXTRACT_SCALAR(report, "$.audits.appcache-manifest.score") AS using_appcache,
+  JSON_EXTRACT_SCALAR(report, "$.audits.service-worker.score") AS using_serviceworkers,
+  COUNT(0) AS occurrences,
+  SUM(COUNT(0)) OVER () AS total,
+  COUNT(0) / SUM(COUNT(0)) OVER () AS pct
+FROM 
+  `httparchive.lighthouse.2019_07_01_mobile` 
+GROUP BY 
+  http_type, 
+  using_appcache, 
+  using_serviceworkers

```

Comment:
Are you explicitly looking for the relationship between these two scores? Would it be clearer to separate them into independent queries?

Reply:
- we wanted to extract the independent or simultaneous occurrence (not the relationship) of protocol (http/https), appcache (used/not used), and service workers (used/not used); theoretically 2 x 2 x 2 = 8 combinations; but ended up with 11 combinations (some of the appcache and service workers had a null value). 
- it may also be cost effective, as the query reads 1.8 TB,